src/PlaybackLooping.cpp

```diff
@@ -0,0 +1,208 @@
+/*!********************************************************************
+*
+ Audacity: A Digital Audio Editor
+
+ PlaybackLooping.cpp
+
+ Anton Gerasimov
+
+ **********************************************************************/
+
+#include "PlaybackLooping.h"
+
+#include "Theme.h"
+#include "AllThemeResources.h"
+
+#include "ZoomInfo.h"
+#include "HitTestResult.h"
+#include "RefreshCode.h"
+#include "TrackPanelMouseEvent.h"
+#include "SelectUtilities.h"
+#include "Track.h"
+
+#include <wx/event.h>
+
+PlaybackLooping::PlaybackLooping()
+{
+	;
+}
+
+PlaybackLooping& PlaybackLooping::GetInstance()
+{
+	static PlaybackLooping instance;
```

Comment:
Why a static instance?  Why not per-project state?

Reply:
I want to isolate all playback looping logic (toggle-on, etc) into PlaybackLooping service, therefore static allow me using this service instance from toolbar button handlers.

Reply:
I do not accept this answer.